File: ployglots/mta-ployglot-pkg-fetch-client/packages/py/examples/api/01_create_async_client.py
"""API example: tuned AsyncClient defaults.

Shows how to construct an `AsyncClient` with:
  - custom Timeout
  - BearerAuth
  - RetryConfig with full jitter

Run:
    python -m examples.api.01_create_async_client
"""

from __future__ import annotations

import asyncio
import sys

from fetch_http_client import (
    AsyncClient,
    BearerAuth,
    JitterStrategy,
    RetryConfig,
    Timeout,
)


async def main() -> None:
    async with AsyncClient(
        base_url="https://httpbin.org",
        auth=BearerAuth("dummy-token-never-used"),
        timeout=Timeout(connect=5.0, read=30.0, write=30.0, pool=5.0),
        retry=RetryConfig(
            max_attempts=3,
            base_delay=0.25,
            max_delay=5.0,
            jitter=JitterStrategy.FULL,
        ),
    ) as client:
        resp = await client.get("/anything")
        resp.raise_for_status()
        sys.stdout.write(f"status={resp.status_code}\n")


if __name__ == "__main__":
    asyncio.run(main())

File: ployglots/mta-ployglot-pkg-fetch-client/packages/ts/examples/api/01-create-async-client.ts
// @ts-nocheck
/**
 * Example: tuned `AsyncClient` defaults for an LLM-style API.
 *
 * `@polyglot/fetch-http-client` exposes `AsyncClient` directly (vs. a thin
 * `createFetchHttpxClient` factory). The constructor accepts the full
 * `AsyncClientOptions` shape — auth, timeout, retry, circuitBreaker, eventHooks,
 * etc. — so most "factory with defaults" patterns collapse to a single ctor call
 * with named fields.
 *
 * Defaults illustrated:
 *   - connect/read timeouts via `Timeout` shorthand.
 *   - followRedirects on with maxRedirects bumped to 10.
 *   - bearer auth pulled from env (LLM_API_KEY).
 *   - retry on 429/5xx with full jitter.
 */
import { AsyncClient, BearerAuth, JitterStrategy } from '@polyglot/fetch-http-client';

async function main(): Promise<void> {
  const client = new AsyncClient({
    timeout: { connect: 5000, read: 120_000, write: 30_000 },
    auth: new BearerAuth(process.env.LLM_API_KEY ?? 'sk-test'),
    headers: { 'user-agent': 'fetch-http-client-example/1.0' },
    followRedirects: true,
    maxRedirects: 10,
    retry: {
      maxRetries: 3,
      retryDelay: 1000,
      retryBackoff: 2,
      jitter: JitterStrategy.FULL,
      retryOnStatus: [429, 502, 503, 504],
    },
  });
  try {
    const resp = await client.post('https://api.example.com/v1/chat', {
      json: { model: 'example-2', messages: [{ role: 'user', content: 'hello' }] },
    });
    console.log('status:', resp.statusCode);
    console.log('body:', await resp.text());
  } finally {
    await client.close();
  }
}

main().catch((err: unknown) => {
  console.error(err);
  process.exitCode = 1;
});
